src/plugin.ts

```diff
@@ -0,0 +1,157 @@
+import { createWriteStream, promises as fs } from 'fs'
+import path from 'path'
+import pixelMatch from 'pixelmatch'
+import { PNG } from 'pngjs'
+import sanitize from 'sanitize-filename'
+import { serializeError, type ErrorObject } from 'serialize-error'
+
+import { createFolder } from './utils/fs'
+import { adjustCanvas, parseImage } from './utils/image'
+import { logger } from './logger'
+
+let CYPRESS_SCREENSHOT_DIR = 'cypress/screenshots' // TODO why ?!
+
+export type UpdateSnapshotArgs = {
+  screenshotName: string
+  specRelativePath: string
+  specFolder: string
+  screenshotsFolder: string
+  snapshotBaseDirectory: string
+}
+
+function getSpecTrimmedPath(relativePath: string, integrationFolder: string): string {
```

Comment:
This part is the most troubling one:
 - two things to consider:
    1) path to specFolder (previously called integrationFolder (here still to be updated))
    2) spec nested positions (see sub/deep.spec.ts test file)
 These are different in run/open mode which drive me nuts, plus specFolder cannot be simply taken from specPattern as it's not a path
I've tried many different things just to end up with some edge case problems... I'll have to see if a spec absolute path can be used in a good way